### ISSUE TITLE
[CoE Starter Kit - QUESTION] what is the correct way to switch connection accounts?

### ISSUE THREAD

**Issue by DaveTheTeamsGuy** [microsoft/coe-starter-kit#2143]
### What is your question?

I originally set up the starter kit with an admin account.  I would now like to switch the connectors over to using a permanent service account.  Using "Switch Account" simply changes the account to the current logged in user.  If I give the new service account admin access to the solution and log in with that to switch the account, I do not see any connectors listed in the environment.  Is there a documented way to switch the starter kit over to using a different account?  Also, will I need to redo all of the run-only user properties to use the new connection account?  Thanks.

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

3.31

### What app or flow are you having the issue with?

Connections

**Comment by osbornecc** [comment]
Hi @DaveTheTeamsGuy 
I was just in here entering my own question and I saw yours.  We actually just did this exact thing a few months ago.  Our CoE was running as my account and we got a permanent service account.

We just run Core Components and Audit Logs for now.  To make sure absolutely everything was running under the new service account, we created a new CoE and deleted the old one.  We created a new environment and reinstalled the solutions as that service account.  Then to make sure we didn't lose our audit log data, we did the steps in #1507 

Once we were satisfied with the new data, we deleted the original CoE environment.  It's a bit challenging, but better than trying to force ownership changes on objects that don't handle that well (like the custom connector for audit logs)

Hope that helps.

**Comment by DaveTheTeamsGuy** [comment]
Thanks @osbornecc I'll take a look at this.  I am ultimately hoping for a switch connection solution, but as I think about it, there are a few things in my original install that don't leave me with a great feeling, so as long as I can preserve audit log data that's probably key.  I did make some modifications to the PBI report to add slicers and extra visualizations to some of the reports, were you also able to move your report over to a new dataset, or did you point the old dataset at the new CoE?

**Comment by osbornecc** [comment]
Honestly, we don't modify the delivered PBI report so we just re-opened it and pointed it to the new URL.  Which I guess is effectively changing the dataset. :-)

Then for our custom reports, I just edited those and updated the URL there, too.

**Comment by manuelap-msft** [comment]
Thank you @osbornecc for jumping in and sharing your experience with this - I really appreciate it and love it when the community helps each other!

The approach @osbornecc took is certainly valid and would set you up with a clean ownership of all objects - where ownership is determined by solution import. 

The other approach would be to update all the connection references to use the new connection. For this
1. Navigate to make.powerapps.com > Solutions
2. Open the Default Solution
3. Filter to only see Connection References
4. Look for all connection references starting with "CoE..." 
5. Click on the connection reference to update it to the new connection

You then may also want to change ownership of apps and flows to be owned by the new account. I think you should be able to do this with the Set Flow Permissions and Set App Permissions app in the CoE kit itself.

**Comment by manuelap-msft** [comment]
If you choose to create a new environment, you should be able to update the OrgUrl paramater in the Power BI report to point to the new environment and that should be pretty seamless unless you have changed the Dataverse tables as well (e.g. added new fields directly in Dataverse). If you've just made changes to the Power BI in terms of filters or how data is shown, that should just work by updating the OrgUrl.

**Comment by DaveTheTeamsGuy** [comment]
I'll take the new install route.  I need to document the install anyway.  Thanks for the assist!

**Comment by DaveTheTeamsGuy** [comment]
Sorry to necro this one, but I just got to moving to a new environment and ran into an issue - 

I'm attempting an audit log import from an old CoE instance to new CoE instance. I have exported the old audit log table to excel, and when I attempt to import it in the new instance (Data > Get Data > Get Data from Excel), everything maps OK, but when I run the import, it runs for a while then I get the following message: "Import failed: There was an error when getting the project execution status. Executing"

I've tried twice, same message both times. I checked dataflows (as suggested at [this link](https://powerusers.microsoft.com/t5/Microsoft-Dataverse/Import-failed-There-was-an-error-when-getting-the-project/td-p/742590) thinking I might see an error, but there's no dataflows listed. Also, I'm guessing since I imported twice, I may have just doubled up the data in the new table.  Finally, it seems that it died before it imported everything because there's a definite mismatch in the number of app launches between the old and new CoE instances.  Thoughts on how to proceed?

**Comment by DaveTheTeamsGuy** [comment]
Based on the data I'm seeing, I think I'm in a good state.  When I uploaded the file, prior to hitting import, everything mapped correctly (got the green check mark).  I'm just wondering about the "Import failed: There was an error when getting the project execution status. Executing" error after the actual import.  This was a brand new CoE instance and a fairly simple export - import operation.

**Comment by Jenefer-Monroe** [comment]
I've seen that map correctly UX be incorrect. I always open it up to sanity check it didnt map to the wrong fields. Just as an FYI on what I do :)

**Comment by DaveTheTeamsGuy** [comment]
I checked the mapping, everything appears to match up.  The only thing not mapped is App Lookup (Not set).

**Comment by Jenefer-Monroe** [comment]
Correct, the app lookup is new so it wont be in your old data. You could manually set this with the excel addin by pasting the string value of the GUID into the lookup column.

**Comment by DaveTheTeamsGuy** [comment]
So that additional column is likely not what caused the original "Import failed: There was an error when getting the project execution status. Executing" message on import.  Hoping at this point it's benign.

**Comment by Jenefer-Monroe** [comment]
A colleague just make this video to help with Audit Log migration using Dataflows. 
https://youtu.be/PL_J1eI5pFA
